OTAnalytics/application/use_cases/highlight_intersections.py

```diff
@@ -115,3 +116,88 @@ def get_ids(self) -> Iterable[TrackId]:
             if assignment.assignment.id in self._flow_state.selected_flows.get():
                 ids.add(TrackId(assignment.road_user))
         return ids
+
+
+class TracksOverlapOccurrenceWindow(TrackIdProvider):
+    def __init__(
+        self,
+        track_repository: TrackRepository,
+        track_view_state: TrackViewState,
+        other: Optional[TrackIdProvider] = None,
+    ) -> None:
+        """Returns track ids that overlap with the current date range filter.
+
+        Args:
+            track_repository (TrackRepository): the track repository.
+            track_view_state (TrackViewState): contains information of current
+                filter element.
+            other (Optional[TrackIdProvider], optional): Takes ids of others for
+                filtering. Defaults to None.
+        """
+        self._other = other
+        self._track_repository = track_repository
+        self._track_view_state = track_view_state
+
+    def get_ids(self) -> Iterable[TrackId]:
+        if self._other:
+            track_ids = self._other.get_ids()
+
+            tracks: list[Track] = []
+            for track_id in track_ids:
+                if track := self._track_repository.get_for(track_id):
+                    tracks.append(track)
+        else:
+            tracks = self._track_repository.get_all()
+
+        return self._filter(tracks)
+
+    def _filter(self, tracks: Iterable[Track]) -> Iterable[TrackId]:
+        date_range = self._track_view_state.filter_element.get().date_range
+        start_date_filter = date_range.start_date
+        end_date_filter = date_range.end_date
+
+        match (start_date_filter, end_date_filter):
+            case (datetime() as start_date_filter, datetime() as end_date_filter):
+                return [
+                    track.id
+                    for track in tracks
+                    if self._has_overlap(
+                        start_date_filter,
+                        end_date_filter,
+                        track.detections[0].occurrence,
```

Comment:
This gets some details of the track. It might be better to add a `start` and `end` method to the `Track` class to encapsulate this behaviour.